FragmentNavigation/src/main/java/com/infomaniak/core/fragmentnavigation/FragmentNavigationExt.kt

```diff
@@ -0,0 +1,61 @@
+/*
+ * Infomaniak Core - Android
+ * Copyright (C) 2025 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.core.fragmentnavigation
+
+import android.os.Bundle
+import androidx.annotation.IdRes
+import androidx.fragment.app.Fragment
+import androidx.navigation.NavController
+import androidx.navigation.NavDirections
+import androidx.navigation.NavOptions
+import androidx.navigation.Navigator
+import androidx.navigation.fragment.DialogFragmentNavigator
+import androidx.navigation.fragment.FragmentNavigator
+import androidx.navigation.fragment.findNavController
+
+fun Fragment.isAtInitialDestination(substituteClassName: String? = null): Boolean {
+    return findNavController().isAtInitialDestination(allowedInitialClassName = substituteClassName ?: javaClass.name)
+}
+
+fun NavController.isAtInitialDestination(allowedInitialClassName: String): Boolean {
+    val currentlyAtClassName = when (val currentlyAtDestination = currentDestination) {
+        is FragmentNavigator.Destination -> currentlyAtDestination.className
+        is DialogFragmentNavigator.Destination -> currentlyAtDestination.className
+        null -> return true
+        else -> return false
+    }
+
+    return allowedInitialClassName == currentlyAtClassName
+}
+
+fun Fragment.safelyNavigate(directions: NavDirections) = findNavController().let { navController ->
```

Comment:
Why not using a `with` here ?  we're totally in a `with` usecase

Reply:
Is it normal that this functions doesn't check for `isAtInitialDestination` ? 
Was the original code's check useless in all cases ?

Reply:
When `currentClassName` is provided `canNavigate(currentClassName)` would simplify to `true`. And when it isn't provided, it would compute the same kind of check as `currentDestination?.getAction(directions.actionId) != null`.

This last bit of code checks whether you're currently at the destination where the action was defined from.

This renders any impersonification impossible which used to be possible when it was only `canNavigate(currentClassName)`. But this new check has been there for a while and it apparently never caused any issue so I guess impersonification isn't needed when dealing with NavDirections, which seems legit.

The new check has been added to fix some kind of IllegalStateException as well apparently, so I kept that new one over the original `canNavigate()`

Reply:
I went with a `let` over the previous `with` because when trying to understand this code with @sirambd, we failed to see what the navController was needed for when there was a `with`, so I tried making it more explicit using a `let`. 

You'd still prefer using a `with` here?

Reply:
Ok sure, I went with the `with` then